### PR TITLE
Update gopkg.in/yaml.v3

### PR DESCRIPTION
**Pull request by sousandrei**
#### References to other Issues or PRs
Addresses #2728 
Depends on https://github.com/stretchr/testify/pull/1192 to completely remove `gopkg.in/yaml.v2`

#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md)?
Yes!

#### Brief description of what is fixed or changed

Updating `gopkg.in/yaml.v2` to `gopkg.in/yaml.v3` to address  [CVE-2022-28948](https://nvd.nist.gov/vuln/detail/CVE-2022-28948)
In the process, bumping `testify` to complete remove `gopkg.in/yaml.v2` and removing `sigs.k8s.io/yaml` for parsing YAML

#### Other comments

Not sure functionality from `sigs.k8s.io/yaml` was strictly needed, if yes we can adjust this PR to include that again, although that will require some rework on said library to support `gopkg.in/yaml.v3`

Waiting for the `testify` PR to be merged to update this pull

### CODE REVIEW

File: internal/descriptor/grpc_api_configuration.go [hunk 1]
```diff
@@ -6,23 +6,13 @@ import (
...
+	var serviceConfiguration apiconfig.GrpcAPIService
+	err := yaml.Unmarshal(yamlFileContents, &serviceConfiguration)
 	if err != nil {
-		return nil, fmt.Errorf("failed to convert gRPC API Configuration from YAML in '%v' to JSON: %v", yamlSourceLogName, err)
-	}
-
-	// As our GrpcAPIService is incomplete, accept unknown fields.
-	unmarshaler := protojson.UnmarshalOptions{
-		DiscardUnknown: true,
-	}
```
Comment (johanbrandhorst) [inline internal/descriptor/grpc_api_configuration.go:22]: I think this was the reason we were using the other YAML parser, so we definitely still need this behavior. Can you make a test that checks that we can still unmarshal unknown fields?

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]

File: internal/descriptor/grpc_api_configuration.go (hunk 1 above)
Comment (sousandrei) [reply internal/descriptor/grpc_api_configuration.go]: You were right on the gold! Totally missed the protojson Marshalling
Should be happy now

File: internal/descriptor/grpc_api_configuration_test.go [hunk 2]
```diff
@@ -67,8 +67,8 @@ http:
 		t.Fatal(err)
 	}
 
-	if !strings.Contains(err.Error(), "line 7") {
-		t.Errorf("Expected yaml error to be detected in line 7. Got other error: %v", err)
+	if !strings.Contains(err.Error(), "line 6") {
```
Comment (johanbrandhorst) [inline internal/descriptor/grpc_api_configuration_test.go:70]: Do we know why this changed?

File: internal/descriptor/grpc_api_configuration_test.go [hunk 3]
```diff
@@ -147,3 +147,42 @@ http:
...
 	}
 }
+
+func TestLoadGrpcAPIServiceFromYAMLUnknownKeys(t *testing.T) {
+	service, err := loadGrpcAPIServiceFromYAML([]byte(`
+type: google.api.Service
+config_version: 3
+
+very: key
+much: 1
```
Comment (johanbrandhorst) [inline internal/descriptor/grpc_api_configuration_test.go:157]: 😂

**Review by johanbrandhorst — APPROVED** [review]

File: protoc-gen-openapiv2/internal/genopenapi/types.go [hunk 4]
```diff
@@ -257,16 +257,12 @@ type keyVal struct {
...
 
 func (p openapiSchemaObjectProperties) MarshalYAML() (interface{}, error) {
-	ms := make(yaml.MapSlice, len(p))
-
-	for i, v := range p {
-		ms[i] = yaml.MapItem{
-			Key:   v.Key,
-			Value: v.Value,
-		}
+	yamlContent, err := yaml.Marshal(p)
```
Comment (johanbrandhorst) [inline protoc-gen-openapiv2/internal/genopenapi/types.go:260]: I think this is going to cause an infinite loop 😂

**Review by johanbrandhorst — CHANGES_REQUESTED** [review]
Tests bork

File: protoc-gen-openapiv2/internal/genopenapi/types.go (hunk 4 above)
Comment (sousandrei) [reply protoc-gen-openapiv2/internal/genopenapi/types.go]: Can't believe I missed that :facepalm:

File: internal/descriptor/grpc_api_configuration_test.go (hunk 2 above)
Comment (sousandrei) [reply internal/descriptor/grpc_api_configuration_test.go:70]: The error message changed to 

```
yaml: line 6: did not find expected key
```

Comment (johanbrandhorst) [reply internal/descriptor/grpc_api_configuration_test.go:70]: Haha, yeah, I can kind of infer that, I was just curious why the line number had changed from 7 to 6.

**Pull request merged**